time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateTimeArbitrary.java

```diff
@@ -101,15 +102,14 @@ private LocalDateTime calculateEffectiveMinDate(LocalDateTime effective) {
 	private LocalDateTime calculateEffectiveMinWithPrecision(LocalDateTime effective) {
 		LocalDate date = effective.toLocalDate();
 		LocalTime time = effective.toLocalTime();
-		try {
-			time = DefaultLocalTimeArbitrary.calculateEffectiveMinWithPrecision(time, ofPrecision);
-		} catch (IllegalArgumentException e) {
+		time = DefaultLocalTimeArbitrary.calculateEffectiveMinWithPrecisionFromOtherClass(time, ofPrecision);
+		if (time == null) {
 			time = LocalTime.MIN;
 			LocalDate effectiveDate;
-			try {
+			if (date.getYear() < 999_999_999 || date.getMonth() != DECEMBER || date.getDayOfMonth() < 31) {
 				effectiveDate = date.plusDays(1);
-			} catch (DateTimeException dateTimeException) {
-				throw e;
+			} else {
+				throw new IllegalArgumentException("Cannot use this min value with precision " + ofPrecision.get());
 			}
 			date = effectiveDate;
 		}
```

Comment:
Let's revisit this code together. IMO it does not communicate well what's happening here.